models/_sector/nft/trades/ethereum/nft_ethereum_base_trades.sql

```diff
@@ -0,0 +1,73 @@
+{{ config(
+    schema = 'nft_ethereum',
+    alias = 'base_trades',
+    tags = ['dunesql'],
+    materialized = 'view'
+    )
+}}
+
+
+{% set nft_models = [
+ ref('archipelago_ethereum_base_trades')
+,ref('superrare_ethereum_base_trades')
+,ref('foundation_ethereum_base_trades')
+,ref('blur_ethereum_base_trades')
+,ref('blur_seaport_ethereum_base_trades')
+,ref('blur_v2_ethereum_base_trades')
+,ref('element_ethereum_base_trades')
+,ref('x2y2_ethereum_base_trades')
+,ref('zora_v1_ethereum_base_trades')
+,ref('zora_v2_ethereum_base_trades')
+,ref('zora_v3_ethereum_base_trades')
+,ref('cryptopunks_ethereum_base_trades')
+,ref('sudoswap_ethereum_base_trades')
+,ref('collectionswap_ethereum_base_trades')
+,ref('looksrare_v1_ethereum_base_trades')
+,ref('looksrare_v2_ethereum_base_trades')
+,ref('looksrare_seaport_ethereum_base_trades')
+] %}
+
+with base_union as (
+SELECT * FROM  (
+{% for nft_model in nft_models %}
+    SELECT
+        blockchain,
+        project,
+        project_version,
+        cast(date_trunc('day', block_time) as date) as block_date,
+        cast(date_trunc('month', block_time) as date) as block_month,
+        block_time,
+        block_number,
+        tx_hash,
+        project_contract_address,
+        trade_category,                 --buy/sell/swap
+        trade_type,                     --primary/secondary
+        buyer,
+        seller,
+        nft_contract_address,
+        nft_token_id,
+        nft_amount,                -- always 1 for erc721
+        price_raw,
+        currency_contract,
+        platform_fee_amount_raw,
+        royalty_fee_amount_raw,
+        platform_fee_address,   -- optional
+        royalty_fee_address,    -- optional
+        sub_tx_trade_id,
+--        tx_from,              -- not yet available in the base event tables
+--        tx_to,                -- not yet available in the base event tables
+        row_number() over (partition by tx_hash, sub_tx_trade_id order by tx_hash) as duplicates_rank   -- duplicates protection
+    FROM {{ nft_model }}
+    {% if is_incremental() %}
+    WHERE {{incremental_predicate('block_time')}}
+    {% endif %}
+    {% if not loop.last %}
+    UNION ALL
+    {% endif %}
+    {% endfor %}
+    )
+where duplicates_rank = 1
```

Comment:
with this being a view, should we remove incremental logic / duplicates check?

Reply:
hmm, good question. We should not remove the duplicate check, but good idea to clean up the incremental blocks.